### DISCUSSION TITLE
Git Sync - all token sets get generated in a single file

### DISCUSSION THREAD

**Discussion by nareshbhatia** [six7/figma-tokens#520] [Q&A] [answered]
The GitHub sync tab asks for a single file name. Is that the intended behavior? I was expecting it to ask for a folder name so that each token set gets output in a separate file. A bit confused because the examples show a separate file for each token set. For example, 4 files are generated in this example: https://github.com/six7/figma-tokens-example-multi/tree/main/tokens

**Comment by nareshbhatia** [comment, marked as answer]
Found the answer to this question [here](https://figma-tokens-docs-git-git-sync-beta-six7.vercel.app/sync/github#7-how-to-use-tokens-stored-in-github-in-development). The plugin indeed generates a single file which needs to be transformed using the [Token Transformer](https://www.npmjs.com/package/token-transformer).

**Reply by six7** [reply]
For now, at least. We're working on supporting multiple files/folders, but right now a single json is the only way unfortunately.

**Reply by nareshbhatia** [reply]
Thanks, not a big deal for now.
